dotnet-datetime/CheckIfDateIsLessThanOrEqualToToday/CheckIfDateIsLessThanOrEqualToToday/CheckDateMethods.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Globalization;
+
+namespace CheckIfDateIsLessThanOrEqualToToday;
+
+public class CheckDateMethods
+{
+    public readonly DateTime _dateToCheck;
+
+    public CheckDateMethods(string dateString = "01/01/2024")
+    {
+        if (!DateTime.TryParseExact(dateString, "MM/dd/yyyy", CultureInfo.InvariantCulture, DateTimeStyles.None, out _dateToCheck))
```

Comment:
Note that the DateTime object that we are creating here, is of an unspecified kind, whereas `DateTime.Today` is `Local`. Whenever performing DateTime comparisons we should ensure that we are comparing like kinds:
https://learn.microsoft.com/en-us/dotnet/api/system.datetimekind?view=net-8.0

Reply:
Hi @jshergal, is there any documentation I can read to get more information on this statement, 'Whenever performing `DateTime` comparisons, we should ensure that we are comparing like kinds'?

The link you provided states that the `DateTimeKind` members are not used in comparisons.

Reply:
Hi @cjaynjoku, you are correct. They are not considered when making comparisons and that is exactly the problem. If for instance I give you a local time in the timezone UTC+10, and you call "DateTime.Today" when the code is running in UTC-10, what is the expected behavior? For example, if in UTC+10 the time is currently 09:00 on 2024.02.08, in the UTC-10 timezone the time would be 13:00 on 2024.02.07, but if we computed the time in UTC they are the same day.
That is why we need to consider the kind when performing comparisons. If we ignore it, then we should explain why we are ignoring it and what the consequences are. It is generally considered a good practice to perform date/time comparisons in UTC for just this exact reason.

Reply:
Alright @jshergal. I get your point now. Thanks for pointing this out.